.github/workflows/ci-build-and-test.yml

```diff
@@ -30,11 +30,12 @@ jobs:
       - name: Run build
         run: npm run build
 
```

Comment:
Let's change the trigger to
```
on:
  pull_request:
    types: [opened]
  workflow_dispatch:
```
The workflow can be retriggered manually if needed.
And make sure merging is not allowed if checks don't pass